buzsaki_lab_to_nwb/petersen_code/petersenmiscdatainterface.py

```diff
@@ -0,0 +1,176 @@
+        metadata_dict: dict,
+        stub_test: bool = False,
+     ):
+        session_path = Path(self.source_data['folder_path'])
+        session_id = session_path.name
+
+        # Trials
+        take_file_paths = [x for x in session_path.iterdir() if "Take" in x.name]
+        # Some sessions had duplicate/non-corresponding Take files
+        if len(take_file_paths) == 1:
+            take_file_path = take_file_paths[0]
+            take_file = pd.read_csv(take_file_path, header=5)
+            take_file_time_name = [x for x in take_file if "Time" in x][0]  # Can be either 'Time' or 'Time (Seconds)'
+            take_frame_to_time = {x: y for x, y in zip(take_file['Frame'], take_file[take_file_time_name])}
+
+            trial_info = loadmat(str(session_path / f"{session_id}.trials.behavior.mat"))['trials']
+            trial_start_frames = trial_info['start'][0][0]
+            n_trials = len(trial_start_frames)
+            trial_end_frames = trial_info['end'][0][0]
+            trial_stat = trial_info['stat'][0][0]
+            trial_stat_labels = [x[0][0] for x in trial_info['labels'][0][0]]
+            cooling_info = trial_info['cooling'][0][0]
+            cooling_map = dict({0: "Cooling off", 1: "Pre-Cooling", 2: "Cooling on", 3: "Post-Cooling"})
+            trial_error = trial_info['error'][0][0]
+            error_trials = np.array([False]*n_trials)
+            error_trials[np.array(trial_error).astype(int)-1] = True  # -1 from Matlab indexing
+
+            trial_starts = []
+            trial_ends = []
+            trial_condition = []
+            for k in range(n_trials):
+                trial_starts.append(take_frame_to_time[trial_start_frames[k]])
+                trial_ends.append(take_frame_to_time[trial_end_frames[k]])
+                nwbfile.add_trial(start_time=trial_starts[k], stop_time=trial_ends[k])
+                trial_condition.append(trial_stat_labels[int(trial_stat[k])-1])
+
+            nwbfile.add_trial_column(
+                name='condition',
+                description="Whether the maze condition was left or right.",
+                data=trial_condition
+            )
+            nwbfile.add_trial_column(
+                name='error',
+                description="Whether the subject made a mistake.",
+                data=error_trials
+            )
+
+            if 'temperature' in trial_info:  # Some sessions don't have this for some reason
+                trial_temperature = trial_info['temperature'][0][0]
+                nwbfile.add_trial_column(
+                    name='temperature',
+                    description="Average brain temperature for the trial.",
+                    data=trial_temperature
+                )
+
+            if len(cooling_info) == n_trials:  # some sessions had incomplete cooling info
+                trial_cooling = []
+                for k in range(n_trials):
+                    trial_cooling.append(cooling_map[int(cooling_info[k])])
```

Comment:
I'd probably make this a list comprehension, but it's not a big deal

Reply:
Fixed - agreed, it takes less space and is more efficient. The for loop was just a copy/paste separating it from above, where a for loop makes sense for constructing the trial table.